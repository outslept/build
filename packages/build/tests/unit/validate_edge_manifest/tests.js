import { test, expect } from 'vitest'

import { validateEdgeFunctionsManifest } from '../../../lib/plugins_core/edge_functions/validate_manifest/validate_edge_functions_manifest.js'
import { removeErrorColors } from '../../../src/error/colors.js'

test('should validate valid manifest', async () => {
  const manifest = {
    bundles: [
      {
        asset: 'f35baff44129a8f6be7db68590b2efd86ed4ba29000e2edbcaddc5d620d7d043.js',
        format: 'js',
      },
    ],
    routes: [
      {
        function: 'hello',
        pattern: '^/hello/?$',
      },
      {
        function: 'geolocation',
        pattern: '^/geolocation/?$',
      },
      {
        function: 'json',
        pattern: '^/json/?$',
      },
    ],
    bundler_version: '1.6.0',
  }

  await expect(validateEdgeFunctionsManifest(manifest)).resolves.not.toThrow()
})

test('should print error on invalid manifest', async () => {
  const manifest = 'json'

  let error
  try {
    await validateEdgeFunctionsManifest(manifest)
  } catch (e) {
    error = e
  }

  expect(error).toBeDefined()
  removeErrorColors(error)
  expect(error.message).toMatchSnapshot()
})

test('should print error on empty manifest', async () => {
  let error
  try {
    await validateEdgeFunctionsManifest({})
  } catch (e) {
    error = e
  }

  expect(error).toBeDefined()
  removeErrorColors(error)
  expect(error.message).toMatchSnapshot()
})
