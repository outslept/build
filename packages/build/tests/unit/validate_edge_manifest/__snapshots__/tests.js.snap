// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should print error on empty manifest 1`] = `
"Validation of Edge Functions manifest failed
REQUIRED must have required property 'bundler_version'

> 1 | {}
    | ^ ☹️  bundler_version is missing here!"
`;

exports[`should print error on invalid manifest 1`] = `
"Validation of Edge Functions manifest failed
TYPE must be object

> 1 | "json"
    | ^^^^^^ 👈🏽  type must be object"
`;
